books/projects/shnf/support.lisp

```diff
@@ -85,12 +106,23 @@
 ;;    (2) A list (POP i p), where i is a nat and p is a SHF
 ;;    (3) A list (POW i p q), where i is a nat and p and q are SHFs
 
-(defun shfp (x)
+(defn shfp (x)
   (if (atom x)
       (integerp x)
     (case (car x)
-      (pop (and (natp (cadr x)) (shfp (caddr x)) (null (cdddr x))))
-      (pow (and (natp (cadr x)) (shfp (caddr x)) (shfp (cadddr x)) (null (cddddr x)))))))
+      (pop (and (consp (cdr x))   (natp (cadr x))
+                (consp (cddr x))  (shfp (caddr x))
+                (null (cdddr x))))
+      (pow (and (consp (cdr x))   (natp (cadr x))
+                (consp (cddr x))  (shfp (caddr x))
+                (consp (cdddr x)) (shfp (cadddr x))
+                (null (cddddr x)))))))
+;(defun shfp (x)
```

Comment:
No, I mean, I have the same comment here as above, i.e., why do you leave the old version here in comments? Why not delete it?